File: algorithms/js/GroupAnagrams/GroupAnagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
 var groupAnagrams = function(strs) {
    const map = new Map();
    for (let str of strs) {
        let ss = str.split('').sort().join('');
        if (map.has(ss)) {
            map.get(ss).push(str)
        } else {
            map.set(ss, [str])
        }
    }
    return [...map.values()];
};

File: algorithms/python/GroupAnagrams/GroupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        m = {}
        for s in strs:
            key = ''.join(sorted(s))
            m[key] = m.get(key, []) + [s]
            
        return m.values()